src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -401,4 +412,22 @@ private static String createPoolStatsInfo(String title, PoolStats poolStats) {
     }
     return title;
   }
+
+  private static Boolean bypassProxy(String account) {
+    String targetHost = account + SNOWFLAKE_DOMAIN_NAME;
+    return System.getProperty(NON_PROXY_HOSTS) != null && isInNonProxyHosts(targetHost);
+  }
+
+  private static Boolean isInNonProxyHosts(String targetHost) {
+    String nonProxyHosts = System.getProperty(NON_PROXY_HOSTS)
+            .replace(".", "\\.")
+            .replace("*", ".*");
```

Comment:
Could you add comments on why the replacement is needed?

Reply:
Is it possible to add tests for this function too?